Handle unavailable state in template fan

thomasdelaet opened the pull request:
## Proposed change

This change introduces support for unavailable state in the template fan.

This fixes issue #31170


## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests


## Additional information

- This PR fixes or closes issue: fixes #31170


## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

thomasdelaet committed:
handle unavailable state in template fan

fixes issue #31170

thomasdelaet committed:
run isort

frenck commented:
This PR changes the behavior in which the state of the template fan becomes "unknown" when the passed in state is unavailable.

This makes we wonder, shouldn't the template fan entity become unavailable in that case as well?

thomasdelaet committed:
if template state is unavailable, entity state is also unavailable

MartinHjelmare commented on homeassistant/components/template/fan.py:
```diff
@@ -363,6 +364,8 @@ async def async_update(self):
             if speed in self._speed_list:
                 self._speed = speed
+            elif speed == STATE_UNAVAILABLE:
+                self._speed = STATE_UNAVAILABLE
```
Speed, oscillating and direction can't be unavailable. Set them to `None`.

thomasdelaet replied:
done. thanks for your comments and help

MartinHjelmare reviewed (commented):
When the template entity state is unavailable we should set `self._available` to `False`.

thomasdelaet committed:
make speed, oscilatting and direction None when state is unavailable

MartinHjelmare commented:
This remains to be resolved:
https://github.com/home-assistant/core/pull/34377#pullrequestreview-438761661

thomasdelaet committed:
set self._available to false if state of template is unavailable

MartinHjelmare commented on homeassistant/components/template/fan.py:
```diff
@@ -341,6 +342,9 @@ async def async_update(self):
         if state in _VALID_STATES:
             self._state = state
+        elif state == STATE_UNAVAILABLE:
+            self._state = state
```
We can remove this line.
```suggestion
```

emontnemery committed:
Support multiple MQTT availability topics (#37418)

* Support multiple MQTT availability topics

* Make availability list and availability_topic exclusive

* Make availability list and availability_topic exclusive

* Add missing abbreviation

MartinHjelmare commented:
Please add tests for the new functionality.
https://github.com/home-assistant/core/blob/dev/tests/components/template/test_fan.py

thomasdelaet commented:
> Please add tests for the new functionality.
> https://github.com/home-assistant/core/blob/dev/tests/components/template/test_fan.py

Thanks for the help @MartinHjelmare . Happy to try but I'm not really familiar with how these tests work. Any guidance you can give on what you expect to see?

MartinHjelmare commented:
Set up the integration, render the relevant states in the template and assert that the correct state is set in the state machine for the corresponding entity. See existing tests for examples.

bsmappee committed:
Smappee dependency update (#37680)

balloob committed:
Fix Hue homekit discovery (#37694)

Co-authored-by: Franck Nijhof <git@frenck.dev>

emontnemery committed:
Tweak MQTT availability (#37719)

emontnemery committed:
Tweak MQTT availability (#37719)

emontnemery committed:
Fix MQTT availability startup race (#37718)

emontnemery committed:
Fix MQTT availability startup race (#37718)

pnbruckner committed:
Add new repeat loop for scripts and automations (#37589)

elupus committed:
Add rfxtrx device classes to known types (#37698)

* Add device classes to known types

* Update homeassistant/components/rfxtrx/sensor.py

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bachya committed:
Re-add ability to use remote files (by URL) in Slack messages (#37161)

* Re-add remote file support for Slack

* More work

* Ensure Slack can only upload files from whitelisted directories

* Cleanup

* Finish work

* Code review

* Messing around

* Final cleanup

* Add comment explaining why we use aiohttp for remote files

* Typo

bdraco committed:
Use the shared zeroconf instance for homekit_controller (#37691)

* Use the shared zeroconf instance for homekit_controller

* bump version

* Update for upstream changes. Thank you @Jc2k !

* naming

* bump version

* empty commit to reset CI as re-run is not working

rohankapoorcom committed:
Add support for the DataUpdateCoordinator to not automatically update (#37734)

rohankapoorcom committed:
Add support for the DataUpdateCoordinator to not automatically update (#37734)

balloob committed:
Uninstall typing (#37735)

pnbruckner committed:
Remove legacy script mode and simplify remaining modes (#37729)

homeassistant committed:
[ci skip] Translation update

prystupa committed:
Support Fan domain in Bond integration (#37703)

* Support Fan domain in Bond integration

* Support Fan domain in Bond integration

* Support Fan domain in Bond integration (apply PR feedback)

* Support Fan domain in Bond integration (apply PR feedback)

prystupa committed:
Fix incorrect comparison of speed "off" by identity instead of by value (#37738)

prystupa committed:
Refactor Bond integration to remove duplication (#37740)

* Refactor Bond integration to remove duplication in Entity code and unit tests

* Refactor Bond integration to remove duplication in Entity code and unit tests (PR feedback)

haemishkyd committed:
Updates to poolsense integration (#37613)

* Created a binary sensor and corrected some review comments.

* Updated the poolsense class and its interface to handle credentials better

* Moved the client session to the PoolSense class.

* Apply suggestions from code review

* Update binary_sensor.py

* Update homeassistant/components/poolsense/__init__.py

* Update sensor.py

* Update binary_sensor.py

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

balloob committed:
Bump ADS to 3.1.3 (#37748)

balloob committed:
Reference constraint files from requirement files (#37751)

* Reference constraint files from requirement files

* Update azure-pipelines-wheels.yml

Co-authored-by: Pascal Vizeli <pvizeli@syshack.ch>

fronzbot committed:
Bump blinkpy version to fix connection errors (#37755)

* Bump blinkpy version to fix connection errors

* Bump blinkpy version to fix connection errors

* Rebased, re-ran gen_requirements_all

fronzbot committed:
Bump blinkpy version to fix connection errors (#37755)

* Bump blinkpy version to fix connection errors

* Bump blinkpy version to fix connection errors

* Rebased, re-ran gen_requirements_all

balloob committed:
Bump pyHS100 to 3.5.1 (#37749)

pnbruckner committed:
Fix script queued mode typo (#37759)

homeassistant committed:
[ci skip] Translation update

prystupa committed:
Upgrade bond-home to 0.0.9 (#37764)

alandtse committed:
Bump teslajsonpy to 0.9.3. (#37771)

Fixes #37684

bdraco committed:
Significantly improve logging performance when no integrations are requesting debug level (#37776)

Kane610 committed:
UniFi - Handle session expiration (#37782)

Kane610 committed:
UniFi - Handle session expiration (#37782)

prystupa committed:
Add Bond hub as a device for bond entities (#37772)

* Introduce Bond Hub concept

* Read Hub version information when setting up entry

* Link entities to Hub using via_device

* Add test to verify created Hub device properties

esev committed:
Add generic unavailable and last_updated metrics for prometheus (#37456)

* Add generic unavailable and last_updated metrics for prometheus

* Updated with feedback from the code review

elupus committed:
Switch rfxtrx to integration level config (#37742)

* Switch to integration level config

* Switch to per device config rather than per entity type

* All roller shutters should be added as covers

(there are non lighting types)

* Fixup tests that used invalid packets for platforms

* Avoid variable re-use

* Allow control events on sensors too

That way we get signal level sensors for these too

* Lint correction

* Don't filter sensors from config

Disable sensors from GUI if the entities are not wanted

* Correct usage of ATTR_ instead of CONF_

* Make sure the logging when a new entity is added includes the event

prystupa committed:
Add support for fan direction in bond integration (#37789)

* Add support for fan direction in bond integration

* Add support for fan direction (PR feedback)

haemishkyd committed:
Apply code quality updates to poolsense (#37781)

* Created a binary sensor and corrected some review comments.

* Updated the poolsense class and its interface to handle credentials better

* Moved the client session to the PoolSense class.

* Apply suggestions from code review

* Update binary_sensor.py

* Update homeassistant/components/poolsense/__init__.py

* Update sensor.py

* Update binary_sensor.py

* Removed unnecessary class variable

* Correcting a merge error.

Co-authored-by: Chris Talkington <chris@talkingtontech.com>

jjlawren committed:
Wrap possible I/O in executor (#37688)

Kane610 committed:
deCONZ - don't let light "attr" events update group data (#37797)

Kane610 committed:
deCONZ - don't let light "attr" events update group data (#37797)

ajschmidt8 committed:
Fix Dockerfile.dev for VS Code devcontainer (#37801)

prystupa committed:
Add basic support for lights in bond integration (#37802)

homeassistant committed:
[ci skip] Translation update

elupus committed:
Replace rfxtrx entity events with integration events  (#37565)

* Drop per entity events

* Correct linting error

* Drop keys()

balloob committed:
Bump aiokafka to 0.6.0 (#37778)

elupus committed:
Drop dummy connection (#37805)

We now mock the module

coolsylver committed:
pydaikin version bump to 2.3.1: (#37682)

* Fix skyfi incorrect setting zone
* Add support for advanced modes for brp069 model

dermotduffy committed:
Allow an extra packet without dts (for Arlo camera streaming) (#37792)

* Allow 1 packet without dts. See https://github.com/twrecked/hass-aarlo/issues/151 .

* Reset boolean once a packet with dts is found.

* Fix no-else-continue lint error.

balloob committed:
Constraints pt3 (#37803)

jschlyter committed:
Add urlencode template filter (#37753)

* add urlencode template filter

* fix flake8

* add test to string ang integer

* better test vectors

elupus committed:
Add rfxtrx ability to send a raw command to device (#37793)

* Add the ability to send a raw command to device

* Add a test for new service

* Use async test

* Sort includes

balloob committed:
Drop white blacklist pt1 (#37816)

balloob committed:
Simplify logger integration (#37780)

Co-authored-by: J. Nick Koston <nick@koston.org>

Commit:
Add devolo binary sensor device class mapping (#37350)

* add device class mapping for binary sensors

* change if else statement to or

engrbm87 committed:
Properly set update_interval during Speedtest setup (#37708)

* Properly set update_interval during setup

* implement new update_interval method

engrbm87 committed:
Properly set update_interval during Speedtest setup (#37708)

* Properly set update_interval during setup

* implement new update_interval method

tizzen33 committed:
Convert Toon expires_in value to float (#37716)

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

prystupa committed:
Apply bond python related feedback from a prior PR (#37821)

elupus committed:
Switch rfxtrx to config entries (#37794)

* Switch to config flow setup

* Add minimal test for config flow

* Add myself as codeowner and address some review concerns

* Address some further review comments

scop committed:
Update Travis-CI to use Python 3.7.1 (#37830)

Refs 89a9634d35c

prystupa committed:
Map bond fan speeds to standard HA speeds (#37808)

haemishkyd committed:
Apply code review changes for poolsense (#37817)

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Fix homekit_controller discovery via zeroconf (#37725)

bdraco committed:
Fix homekit_controller discovery via zeroconf (#37725)

balloob committed:
Bumped version to 0.112.5

balloob committed:
Merge pull request #37836 from home-assistant/rc

kennedyshead committed:
Version bump for asuswrt (#37827)

Co-authored-by: magnusknutas <magnus@thefarm.se>

scop committed:
Travis CI improvements (#37840)

dependabot[bot] committed:
Bump actions/upload-artifact from v2.0.1 to 2.1.0 (#37841)

Bumps [actions/upload-artifact](https://github.com/actions/upload-artifact) from v2.0.1 to 2.1.0.
- [Release notes](https://github.com/actions/upload-artifact/releases)
- [Commits](https://github.com/actions/upload-artifact/compare/v2.0.1...ebad382c0953e8c6b4039e8d30dfd19ee7b2a862)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prystupa committed:
Add support for generic device (switch) to bond integration (#37837)

pnbruckner committed:
Add choose script action (#37818)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

scop committed:
Attrs cleanups (#37849)

bramkragten committed:
Add mode info attributes to script and automation (#37815)

Co-authored-by: J. Nick Koston <nick@koston.org>

aaliddell committed:
Fix media_content_id attribute in Spotify integration (#37853)

Previously, the media_content_id field would contain the track name, which
appears was a regression when the Spotify integration was rewritten in
7e4b9adc. The media_content_id now reverts to containing the Spotify
track 'uri' field, which has the form 'spotify:track:...'

bramkragten committed:
Frontend: deprecate `extra_html_url` (#37843)

bdraco committed:
Switch async_track_state_change to the faster async_track_state_change_event part 3 (#37852)

async_track_state_change_event is faster than async_track_state_change

bramkragten committed:
Adjust icons for MDI bump (#37730)

bdraco committed:
Avoid homekit crash when temperature is clamped above max value (#37746)

frenck committed:
Always expose Toon gas sensors (#37829)

timmo001 committed:
Use size of camera in Agent DVR (#36375)

bdraco committed:
Adjust history as all scripts can now be canceled (#37820)

bdraco committed:
Ensure HomeKit does not throw when a linked motion sensor is removed (#37773)

Commit:
Add HmIP-FSI16 to HomematicIP Cloud (#37715)

danielpervan committed:
Fix Fibaro HC light switches not being configured as Light entities (#37690)

jbeyerstedt committed:
Do no crash Luftdaten on additional data returned by the API (#37763)

dshokouhi committed:
Fix zone cleaning and raise config entry not ready when needed (#37741)

frenck committed:
Merge branch 'master' into dev

dmulcahey committed:
bump zigpy and zha quirks (#37859)

bramkragten committed:
Updated frontend to 20200714.0 (#37862)

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Have async_track_point_in_utc_time call async_run_job directly from call_at (#37790)

We do not need a nested function here since call_at
takes args

prystupa committed:
Add support for fireplaces to bond integration (#37850)

bdraco committed:
Update august manufacturer name (#37867)

* Update august manufacturer name

This allows homekit to link the accessories to the august app

* yeah.. update the test

bdraco committed:
Switch async_track_state_change to the faster async_track_state_change_event part 6 (#37869)

Calling async_track_state_change_event directly is faster than async_track_state_change (see #37251) since async_track_state_change is a wrapper around async_track_state_change_event now

bdraco committed:
Switch async_track_state_change to the faster async_track_state_change_event part 5 (#37866)

Calling async_track_state_change_event directly is faster than async_track_state_change (see #37251) since async_track_state_change is a wrapper around async_track_state_change_event now

bdraco committed:
Switch async_track_state_change to the faster async_track_state_change_event part 4 (#37863)

* Switch async_track_state_change to the faster async_track_state_change_event part 4

Calling async_track_state_change_event directly is faster than async_track_state_change (see #37251) since async_track_state_change is a wrapper around async_track_state_change_event now

* pylint

bdraco committed:
Switch async_track_state_change to the faster async_track_state_change_event (#37834)

async_track_state_change_event is faster than async_track_state_change

bdraco committed:
Switch a few more async_track_state_change to the faster async_track_state_change_event (#37833)

async_track_state_change_event is faster than async_track_state_change
and since we do not care about the data being returned to the callback
this is a simple speedup

bdraco committed:
Switch universal media_player to use async_track_state_change_event (#37832)

async_track_state_change_event is faster than async_track_state_change
and since we do not care about the data being returned to the callback
this is a simple speedup

bdraco committed:
Improve handling of template platforms when entity extraction fails (#37831)

Most of the the template platforms would check for
extract_entities failing to extract entities and avoid
setting up a state change listner for MATCH_ALL after
extract_entities had warned that it could not extract
the entities and updates would need to be done manually.
This protection has been extended to all template platforms.

Alter the behavior of extract_entities to return the
successfully extracted entities if one or more templates
fail extraction instead of returning MATCH_ALL

bdraco committed:
Switch async_track_state_change to the faster async_track_state_change_event part 7 (#37870)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

balloob committed:
Prefer external URLs because internal can't have valid SSL (#37872)

shkertt committed:
Use supervisord "group:name" when get process info (#37678)

frenck committed:
Don't reuse venv cache when Python version changes (#37881)

shenxn committed:
Fix yeelight flash (#37743)

* Fix yeelight flash

* Use cast instead of string comparison

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

pnbruckner committed:
Add tests

pnbruckner committed:
Update per review 3

dshokouhi committed:
Improve Neato error logging by including device name (#37865)

pnbruckner committed:
Stop running scripts at shutdown (#37858)

bramkragten committed:
Updated frontend to 20200715.0 (#37884)

emontnemery committed:
Adapt MQTT config flow to default birth and will (#37875)

bdraco committed:
Provide workaround for missing/disabled/broken IPv6 (#37887)

pvizeli committed:
Revert breaking change for Automation (#37885)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

frenck committed:
Bumped version to 0.113.0b0

bramkragten committed:
Update frontend to 20200715.1 (#37888)

bramkragten committed:
Update frontend to 20200715.1 (#37888)

bdraco committed:
Remove support for legacy logbook events created before 0.112 (#37822)

* Remove support for legacy logbook events created before 0.112

Reduce the complexity of the logbook code.  This
should also have a small performance boost.

* None is the default

prystupa committed:
Refactor bond tests (#37868)

sgryphon committed:
Feature/izone temperature precision (#37669)

* Change to precision tenths for current temp

mdegat01 committed:
Add `ignore_attributes` option to influxdb (#37747)

* Added ignore_attributes option and tests

* adjusted config for overlapping customization with ignore attrs

frenck committed:
Fix swapped variables deprecation in log message (#37901)

frenck committed:
Fix swapped variables deprecation in log message (#37901)

prystupa committed:
Improve bond startup performance (#37900)

bramkragten committed:
Updated frontend to 20200716.0 (#37910)

bramkragten committed:
Updated frontend to 20200716.0 (#37910)

pnbruckner committed:
Fix automation & script restart mode bug (#37909)

pnbruckner committed:
Fix automation & script restart mode bug (#37909)

Adminiuga committed:
Fix ZHA electrical measurement sensor initialization (#37915)

* Refactor cached ZHA channel reads.

If doing a cached ZCL attribute read, do "only_from_cache" read for
battery operated devices only. Mains operated devices will do a network
read in case of a cache miss.

* Use cached attributes for ZHA electrical measurement

* Bump up ZHA zigpy dependency.

Adminiuga committed:
Fix ZHA electrical measurement sensor initialization (#37915)

* Refactor cached ZHA channel reads.

If doing a cached ZCL attribute read, do "only_from_cache" read for
battery operated devices only. Mains operated devices will do a network
read in case of a cache miss.

* Use cached attributes for ZHA electrical measurement

* Bump up ZHA zigpy dependency.

cgtobi committed:
Fix unavailable when value is zero (#37918)

cgtobi committed:
Fix unavailable when value is zero (#37918)

amelchio committed:
Upgrade pysonos to 0.0.32 (#37923)

amelchio committed:
Upgrade pysonos to 0.0.32 (#37923)

bdraco committed:
Ensure a state change tracker setup from inside a state change listener does not fire immediately (#37924)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Ensure a state change tracker setup from inside a state change listener does not fire immediately (#37924)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

balloob committed:
Bumped version to 0.113.0b1

DaAwesomeP committed:
Add Firmata Integration (attempt 2) (#35591)

firstof9 committed:
Add RGB light support to ozw (#37636)

prystupa committed:
Apply feedback on bond integration (#37921)

bdraco committed:
Increase test line coverage of homeassistant/helpers/event.py to 100% (#37927)

* Increase test line coverage of homeassistant/helpers/event.py to 100%

* fix test

bdraco committed:
Cleanup logbook tests to prevent failure on race condition (#37928)

elupus committed:
Rfxtrx use previous received event to do complete restore (#37819)

* Add event attribute to display last received event

* Restore state using event attribute

* Allow empty dict for device config

* Must also validate normal case

* Do early return

SeraphimSerapis committed:
Fix: Passes secure parameter when setting up Nuki (#36844) (#37932)

* Passes secure parameter when setting up Nuki (#36844)

* Adds an additional configuration option for soft bridges instead of passing True when setting up the bridge

* Revert "Adds an additional configuration option for soft bridges instead of passing True when setting up the bridge"

This reverts commit af1d839ab1c130535c5a31ee60e35d3b2c151c5b.

SeraphimSerapis committed:
Fix: Passes secure parameter when setting up Nuki (#36844) (#37932)

* Passes secure parameter when setting up Nuki (#36844)

* Adds an additional configuration option for soft bridges instead of passing True when setting up the bridge

* Revert "Adds an additional configuration option for soft bridges instead of passing True when setting up the bridge"

This reverts commit af1d839ab1c130535c5a31ee60e35d3b2c151c5b.

marcelveldt committed:
Add ozw support for single setpoint thermostat devices (#37713)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

marcelveldt committed:
Add ozw support for single setpoint thermostat devices (#37713)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

abmantis committed:
Change ZHA power unit from kW to W (#37896)

* Change ZHA power unit from kW to W

* Use POWER_WATT

* Move kW to W conversion; ignore unit for power

abmantis committed:
Change ZHA power unit from kW to W (#37896)

* Change ZHA power unit from kW to W

* Use POWER_WATT

* Move kW to W conversion; ignore unit for power

rajlaud committed:
Fix bugs updating state of `hdmi_cec` switch (#37786)

rajlaud committed:
Fix bugs updating state of `hdmi_cec` switch (#37786)

mweinelt committed:
Improve setup script portability (#37935)

dshokouhi committed:
Switch back to create task for Neato (#37913)

mweinelt committed:
prometheus: Reduce loglevel of failed float conversion to debug (#37936)

It creates alot of useless noise currently.

Fixes #30186

Shulyaka committed:
Add humidifier intents (#37335)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

Shulyaka committed:
Add humidifier device conditions (#36962)

devbis committed:
Fix Yandex transport Integration, add signature to requests (#37365)

elupus committed:
Rftrx force update (#37944)

* Make sure sensor and binary_sensor force update

This make sure it's possible to react to events on all updates.

* Correct addition of binary sensors

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Mock out I/O in the default_config test (#37897)

This test never passed locally because of the I/O.

jjlawren committed:
Fix Sonos speaker lookup for Plex (#37942)

jjlawren committed:
Fix Sonos speaker lookup for Plex (#37942)

Shulyaka committed:
Add humidifier device triggers (#36887)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

bdraco committed:
Index entity_registry_updated listeners (#37940)

bdraco committed:
Automatically recover when the sqlite3 database is malformed or corrupted (#37949)

* Validate sqlite database on startup and move away if corruption is detected.

* do not switch context in test -- its all sync

shenxn committed:
Add switch to pi_hole integration (#35605)

Co-authored-by: Ian <vividboarder@gmail.com>

bdraco committed:
Bump zeroconf to 0.28.0 (#37951)

bdraco committed:
fix (#37889)

bdraco committed:
fix (#37889)

elupus committed:
Make sensor and binary_sensor inherit from base class (#37946)

* Make sensor and binary_sensor inherit from base class

* Drop several pointless public properties

* Make sure base function has same parameters

* Drop pass

* Missed one

* Adjust inherit order

dmulcahey committed:
Force updates for ZHA light group entity members (#37961)

* Force updates for ZHA light group entity members

* add a 3 second debouncer to the forced refresh

* lint

dmulcahey committed:
Force updates for ZHA light group entity members (#37961)

* Force updates for ZHA light group entity members

* add a 3 second debouncer to the forced refresh

* lint

elupus committed:
Add support for InputSelector trait (#35753)

knudsvik committed:
Adds median to min_max component (#36686)

elupus committed:
Rfxtrx fixes for beta (#37957)

balloob committed:
Bumped version to 0.113.0b2

homeassistant committed:
[ci skip] Translation update

michaeldavie committed:
Make nested get() statements safe (#37965)

michaeldavie committed:
Make nested get() statements safe (#37965)

emontnemery committed:
Bump pychromecast to 7.1.2 (#37976)

emontnemery committed:
Bump pychromecast to 7.1.2 (#37976)

lawtancool committed:
Add Control4 integration (#37632)

Co-authored-by: J. Nick Koston <nick@koston.org>
Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

dmulcahey committed:
Force updates for ZHA light group entity members (Part 2) (#37995)

dmulcahey committed:
Force updates for ZHA light group entity members (Part 2) (#37995)

homeassistant committed:
[ci skip] Translation update

balloob committed:
Bumped version to 0.113.0b3

dmulcahey committed:
Update comment about parallel updates to match the documentation (#37964)

JeffLIrion committed:
Close androidtv ADB socket connection when Home Assistant stops (#37973)

* Close the ADB connection on HA stop

* Get the test to pass

* Remove unnecessary test code

* Register the callback sooner

* '_async_stop' -> 'async_close'

* 'async_close' method -> '_async_close' function

dshokouhi committed:
Bumpy pyobihai to make last reboot update as needed (#37914)

bdraco committed:
Index the entity registry (#37994)

jnewland committed:
Fix notify.slack service calls using data_template (#37980)

jnewland committed:
Fix notify.slack service calls using data_template (#37980)

bdraco committed:
Index the device registry (#37990)

jyavenard committed:
Don't advertise switch devices as dimmable lights (#37978)

This issue has been corrected and then reverted multiple times.
It seems that the core issue was a casing one (On/off vs On/Off) ; for better
matching with a real Hue hub, also add the productname.

Tested to work with echo 2 and echo 5.

RobBie1221 committed:
Disable polling for ozw entities (#38005)

RobBie1221 committed:
Force updates for ozw sensors (#38003)

* Force updates and disable polling

* Move force_update to sensor

adamkrol93 committed:
Add Wolflink integration (#34104)

* WOLF Smart-set integration

* Removed translations. Changed device class of timestamp. Added new test for unknown exception

* Remove unit_of_measurement from hours sensor

* Code cleanup. Pull Request comments fixes

* ConnectError import change. Removed DEVICE_CLASS_TIMESTAMP

* Add unique id guard with tests. Use common translations. Move device_id resolution to config_flow.

* Remove debug print

dependabot[bot] committed:
Bump codecov/codecov-action from v1.0.10 to v1.0.11 (#38006)

Bumps [codecov/codecov-action](https://github.com/codecov/codecov-action) from v1.0.10 to v1.0.11.
- [Release notes](https://github.com/codecov/codecov-action/releases)
- [Commits](https://github.com/codecov/codecov-action/compare/v1.0.10...6d208f5b527841fb050f92f778e86cb808dacdcb)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dmulcahey committed:
Cleanup async_accept_signal in ZHA (#38009)

teharris1 committed:
Fix issue with Insteon events not firing correctly (#37974)

teharris1 committed:
Fix issue with Insteon events not firing correctly (#37974)

tradiuz committed:
Fix ozw color temp (#38012)

* Fix color temp math

* Ran black --fast

* Update test_light.py

* tweaking mireds

* updating comments

* fixing test_light to match standards

* fixing comments, need coffee

rsnodgrass committed:
Add scrape sensor name to logs (#38020)

homeassistant committed:
[ci skip] Translation update

dshokouhi committed:
Check if robot has boundaries to update (#38030)

dshokouhi committed:
Check if robot has boundaries to update (#38030)

bdraco committed:
Use event loop scheduling for tracking time patterns (#38021)

* Use event loop scheduling for tracking time patterns

* make patching of time targetable

* patch time tests since time can tick to match during the test

* fix more tests

* time can only move forward

* time can only move forward

* back to 100% coverage

* simplify since the event loop time cannot move backwards

* simplify some more

* revert simplify

* Revert "revert simplify"

This reverts commit bd42f232f6f4e0876fbfe7ce5abf363779761acd.

* Revert "simplify some more"

This reverts commit 2a6c57d51487d4fc85140033a5cadbc800e61287.

* Revert "simplify since the event loop time cannot move backwards"

This reverts commit 3b13714ef489b5c5661ae0254b12f137d047fc26.

* Attempt another simplify

* time does not move backwards in the last two

* remove next_time <= now check

* fix previous merge error

RobBie1221 committed:
Move data on import in rfxtrx integration into ConfigEntry (#38022)

* Move all data imported from yaml to ConfigEntry

* Revert changes that prevent updating yaml entry

* Cleanup code around time conversion

dependabot[bot] committed:
Bump actions/setup-python from v2 to v2.1.1 (#38034)

Bumps [actions/setup-python](https://github.com/actions/setup-python) from v2 to v2.1.1.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v2...0c28554988f6ccf1a4e2818e703679796e41a214)

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

scop committed:
Fix wolflink datetime import (#38028)

emontnemery committed:
Correct arguments to MQTT will_set (#38036)

emontnemery committed:
Correct arguments to MQTT will_set (#38036)

emontnemery committed:
Use keywords for MQTT birth and will (#38040)

emontnemery committed:
Use keywords for MQTT birth and will (#38040)

bdraco committed:
Update tests that track time to account for microsecond precision (#38044)

Adminiuga committed:
ZHA dependencies bump bellows to 0.18.0 (#38043)

Adminiuga committed:
ZHA dependencies bump bellows to 0.18.0 (#38043)

RobBie1221 committed:
Implement unload entry for rfxtrx integration (#38037)

* Implement unload entry

* Change async_remove to remove

* Pop data from hass.data

* Change sequence order in unload

* Dont unload internal when unload platforms fail

elupus committed:
Make sure command entities restore from state (#38038)

elupus committed:
Avoid using implementation internal to trigger events (#38041)

This uses the moxking in fixture to trigger events.

homeassistant committed:
[ci skip] Translation update

bdraco committed:
Ensure we do not start discovered flows until after the started event has fired (#38047)

* Ensure we do not start discovered flows until after the start event has fired

This change makes zeroconf and ssdp match discovery behavior of not
creating config flows until the start event has been fired.  This
prevents config flow creation/dependency installs for discovered
config flows from competing for cpu time during startup.

* Start discovery/service browser/ssdp when EVENT_HOMEASSISTANT_STARTED is fired instead of EVENT_HOMEASSISTANT_START

donkawechico committed:
Support default transition in light profiles (#36747)

Co-authored-by: Paulus Schoutsen <paulus@home-assistant.io>

bdraco committed:
Speed up group setup (#38048)

pnbruckner committed:
Allow float values in time periods (#38023)

balloob committed:
Add MQTT to constraints file (#38049)

balloob committed:
Add MQTT to constraints file (#38049)

elupus committed:
Fix rfxtrx stop after first non light (#38057)

elupus committed:
Fix rfxtrx stop after first non light (#38057)

elupus committed:
Rfxtrx fixup restore (#38039)

frenck committed:
Bumped version to 0.113.0

pvizeli committed:
Update home assistant base image (#38063)

scop committed:
Various type hint improvements (#37952)

frenck committed:
Merge pull request #38065 from home-assistant/rc

pnbruckner committed:
Update automation logger to include object_id like scripts (#37948)

frenck committed:
Merge branch 'master' into dev

RogerSelwyn committed:
Change sky_hub to async and fix exception spamming (#37129)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

frenck committed:
Bump version to 0.114.0dev0 (#38071)

DubhAd committed:
Update discord.py to v1.3.4 for API change (#38060)

frenck committed:
Upgrade aiohttp to 3.6.2 (#38082)

RobBie1221 committed:
Fix issue with creation of PT2262 devices in rfxtrx integration (#38074)

MartinHjelmare committed:
Fix ozw light color values check (#38067)

Co-authored-by: Paulus Schoutsen <balloob@gmail.com>

andrewsayre committed:
Bump pysmartthings to v0.7.2 (#38086)

thomasdelaet committed:
add test to check whether unavailability states work

thomasdelaet committed:
Merge branch 'template-fan-fix' of github.com:thomasdelaet/core into template-fan-fix

homeassistant commented:
Hi @dependabot[bot], 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#34377).

Once you do that we will be able to review and accept this pull request.

Thanks!
